src/decode.c

```diff
@@ -0,0 +1,615 @@
+#include <assert.h>
+#include <stdio.h>
+
+#include "decode.h"
+
+static inline bool decode_load(struct rv_insn_t *rv_insn, uint32_t insn)
```

Comment:
Use `op_load` instead of `decode_load` because we would emphasize the handling of RISC-V instruction. Later, when we introduce advanced code generators, the consistent naming scheme would benefit from them.

Reply:
The expected results from RISC-V instruction decoding would be the sequence of intermediate representation (IR), which allows faster execution. Please check the relevant implementation `refactor-rv32` and `fastrv32`.

Reply:
It does not make sense if you only move functions out of `emulate.c` without introducing IR. So-called decoupling must be involved with more abstractions.

Reply:
TODO: use hashtable to map RISC-V instructions (index) to specific handlers.